Automation Examples/010.md

```diff
@@ -0,0 +1,200 @@
+
+```js
+document.getElementsByClassName("normalText")[2].innerText
+document.getElementsByClassName("normalText")[6].innerText
+document.getElementsByClassName("normalText")[10].innerText
+```
+Getting the buttons to open company webpages is quite similar. Here are examples of how you can open the first three of them:
+
+```js
+document.getElementsByClassName("normalText")[3].getElementsByTagName("a")[0].click()
+document.getElementsByClassName("normalText")[7].getElementsByTagName("a")[0].click()
+document.getElementsByClassName("normalText")[11].getElementsByTagName("a")[0].click()
+```
+
+You can notice that the index for each record increases by 4 and that the respective buttons' indexes are greater by 1 than the company names' indexes.
+
+#### 4. Based on your research, create a for loop in which you will wait for the elements to be loaded, get a company name and then open its webpage. 
+
+```G1ANT
+    for ♥i from 2 to 198 step 4
+        selenium.waitforvalue ‴return document.getElementsByClassName("normalText").length > 0‴ expectedvalue true
+        selenium.runscript ‴return document.getElementsByClassName("normalText")[♥i].innerText‴
+        ♥companyName = ♥result
+        selenium.runscript ‴document.getElementsByClassName("normalText")[♥i+1].getElementsByTagName("a")[0].click()‴
+    end for
+```
+
+This for loop will create an `i` variable which will be initialised with 2 and increased by 4 with each iteration of a loop until the `i` variable reaches 198 because we want to get only the first 50 companies (so you can calculate it where 198 number is from: 2+4*49). 
+
+`selenium.waitforvalue` command is used here to wait for "normalText" class elements to be loaded, `selenium.runscript` command runs javascript code that you tested in your browser. The `companyName` variable stores what this command has returned. 
+
+#### 5. After you have opened a company webpage, wait for the elements to be loaded and click the first .pdf file in the table to download it. 
+
+After a quick investigation, you should see that opening the first .pdf file on this webpage is possible by running this script:
+
+```js
+document.getElementsByClassName("date")[2].getElementsByTagName("a")[0].click()
+```
+
+But obviously you remember about waiting for these class elements to be loaded so first you add this line of code:
+
+```G1ANT
+        selenium.waitforvalue ‴return document.getElementsByClassName("date").length > 0‴ expectedvalue true
+```
+
+And then you open the .pdf file like this:
+
+```G1ANT
+        selenium.runscript ‴document.getElementsByClassName("date")[2].getElementsByTagName("a")[0].click()‴
+```
+
+#### 6. The .pdf file has been opened in the second tab so save it under the correct folder (see the process description)
+
+```G1ANT
+        ♥dateformat = yyyy-MM-dd
+        ♥path = ♥basePath\♥date\♥companyName
+        keyboard ⋘CTRL+S⋙
+        ⊂
+        System.IO.Directory.CreateDirectory(♥path);
```

Comment:
don't we have a command for creating directories? maybe add a reference to CreateDirectory documentation here?

Reply:
We really don't have a command like that yet xD @przegwar also made a mistake like this and during our webinar advised someone to use not-existing command for it 😛